Doc: Document non-atomicity of Hive multi-table inserts (#2394)
apache/iceberg: site/docs/hive.md

#### Hive Query Engines
Both the Map Reduce and Tez query execution engines are supported.

#### Hive Multi-table inserts

> It is possible to issue inserts targeting multiple Hive tables backed by Iceberg. The users should be aware that the commits are atomic only on table level, and the commit is not atomic on query level. The commits to Iceberg tables happen 1-by-1, and if one of the commit fails the other writes will not be rolled back. Example:

Reviewer: small typo: `if one of the commits fails`

Author: Fixed

> It is possible to issue inserts targeting multiple Hive tables backed by Iceberg. The users should be aware that the commits are atomic only on table level, and the commit is not atomic on query level. The commits to Iceberg tables happen 1-by-1, and if one of the commit fails the other writes will not be rolled back. Example:

Reviewer 2: I have a slight suggestion here but be free to ignore it, just a rewording.

Instead of "The users should be"

"Multi-table inserts will not be atomic and are committed one table at a time.  Partial changes will be visible during the commit process and failures can leave partial changes committed. Changes within a single table will remain atomic."

But feel free to ignore. I just wanted to move the important part closer to the beginning of the paragraph.

Author: Seems like a much better description than my one. Updated the PR.

Thanks,
Peter

> It is possible to issue inserts targeting multiple Hive tables backed by Iceberg. The users should be aware that the commits are atomic only on table level, and the commit is not atomic on query level. The commits to Iceberg tables happen 1-by-1, and if one of the commit fails the other writes will not be rolled back. Example:

Reviewer 3: ```suggestion
It is possible to issue inserts targeting multiple Hive tables backed by Iceberg. Users should be aware that the commits are atomic only at the _table_ level and not at the _query_ level. The commits to Iceberg tables happen 1-by-1, and if one of the commits fails the other writes will not be rolled back. Example:
```

(or go with @RussellSpitzer's  rewording above)

Author: Went with @RussellSpitzer's suggestion. For me it seems much cleaner than my version.
